scripts/us_eia/opendata/process/coal.py

```diff
@@ -0,0 +1,236 @@
+        'measuredProperty: dcs:sulfurContent',
+        'statType: dcs:meanValue',
+    ],
+}
+
+_CONSUMING_SECTOR = {
+    '1': 'ElectricUtility',
+    '2': 'ElectricUtilityNonCogen',
+    '3': 'ElectricUtilityCogen',
+    '8': 'CommercialAndInstitutional',
+    '9': 'CokePlants',
+    '10': 'OtherIndustrial',
+    '94': 'IndependentPowerProducers',
+    '98': 'ElectricPower',
+}
+
+_UNIT_MAP = {
+    'ASH_CONTENT': ('', '100'),
+    'HEAT_CONTENT': ('BtuPerPound', ''),
+    'SULFUR_CONTENT': ('', '100'),
+    'CONS_TOT': ('ShortTon', ''),
+    'RECEIPTS': ('ShortTon', ''),
+    'STOCKS': ('ShortTon', ''),
+    'COST': ('USDollarPerShortTon', ''),
+}
+
+
+def generate_statvar_schema(raw_sv, rows, sv_map, counters):
+    """Generate StatVar with full schema.
+
+    Args:
+        raw_sv: Raw stat-var returned by extract_place_statvar()
+        rows: List of dicts corresponding to CSV row. See common._COLUMNS.
+        sv_map: Map from stat-var to its MCF content.
+        counters: Map updated with error statistics.
+
+    Returns True if schema was generated, False otherwise.
+    """
+    counters['generate_statvar_schema'] += 1
+
+    # COAL.{Measure}.{ConsumingSector}.{Period}
+    m = re.match(r"^COAL\.([^._]+_?[^._]+)\.([0-9]+)\.([AQM])$", raw_sv)
+    if m:
+        measure = m.group(1)
+        consuming_sector = m.group(2)
+        period = m.group(3)
+    else:
+        counters['error_unparsable_raw_statvar'] += 1
+        return False
+    counters[f'measure-{measure}'] += 1
+
+    # Get popType and mprop based on measure.
+    measure_pvs = _MEASURE_MAP.get(measure, None)
+    if not measure_pvs:
+        counters[f'error_missing_measure-{measure}'] += 1
+        return False
+
+    sv_id_parts = [common.PERIOD_MAP[period], measure_pvs[0]]
+    sv_pvs = measure_pvs[1:] + [
+        'typeOf: dcs:StatisticalVariable',
+        # TODO(shanth): use new property in next iteration
+        f'measurementQualifier: dcs:{common.PERIOD_MAP[period]}',
+    ]
+
+    if consuming_sector:
+        cs = _CONSUMING_SECTOR.get(consuming_sector, None)
+        if not cs:
+            counters[f'error_missing_consuming_sector-{consumingSector}'] += 1
+            return False
+        sv_id_parts.append(cs)
+        sv_pvs.append(f'consumingSector: dcs:{cs}')
+
+    if measure not in _UNIT_MAP:
+        counters[f'error_missing_unit-{measure}'] += 1
+        return False
+    (unit, sfactor) = _UNIT_MAP[measure]
+
+    sv_id = '_'.join(sv_id_parts)
+
+    # Update the rows with new StatVar ID value and additional properties.
+    for row in rows:
+        row['stat_var'] = f'dcid:{sv_id}'
+        if unit:
+            row['unit'] = f'dcid:{unit}'
+        else:
+            # Reset unit to empty to clear the raw unit value.
+            row['unit'] = ''
+        if sfactor:
+            row['scaling_factor'] = sfactor
+
+    if sv_id not in sv_map:
+        node = f'Node: dcid:{sv_id}'
+        sv_map[sv_id] = '\n'.join([node] + sv_pvs)
```

Comment:
Ah... this reminds me that it might be worth refactoring the interface a bit so this happens in common.